omniscidb/QueryEngine/CMakeLists.txt

```diff
@@ -133,6 +133,7 @@ list(APPEND query_engine_source_files
         CostModel/CostModel.cpp
         CostModel/DataSources/EmptyDataSource.cpp
         CostModel/ExtrapolationModels/LinearExtrapolation.cpp
+        CostModel/ExtrapolationModels/LinearRegression.cpp
```

Comment:
I think we should make the cost model live in a separate library. Here you're unconditionally adding the sources to the QueryEngine library, thus leaking the dependency.

Reply:
Then we also want to hide `armadillo` and `DwarfBench` dependencies from headers, am I right?